test/integration/hotrestart_test.sh

```diff
@@ -276,4 +276,17 @@ start_test disabling hot_restart by command line.
 CLI_HOT_RESTART_VERSION=$("${ENVOY_BIN}" --hot-restart-version --disable-hot-restart 2>&1)
 check [ "disabled" = "${CLI_HOT_RESTART_VERSION}" ]
 
+#Validate socket mode
+start_test socket-mode for socket path
+run_in_background_saving_pid "${ENVOY_BIN}" -c "${HOT_RESTART_JSON}" \
+      --restart-epoch 0  --base-id 0 --base-id-path "${BASE_ID_PATH}" \
+      --socket-path /tmp/envoy_domain_socket --socket-mode 644 \
```

Comment:
`${TEST_TMPDIR}/envoy_domain_socket`

Reply:
Exceeds maximum UDS path size.

Reply:
I'm worried that this may cause conflicts between concurrently running variations of the test. @envoyproxy/maintainers any guidance on this?

Reply:
agreed. Try this with --runs_per_test=100.

You could also make this unique I think using BASE_ID_PATH; not sure if that would exceed the limit. What is the limit?

Reply:
Its limit is 108 bytes.

Reply:
Done.